cwms_radar_api/src/main/java/cwms/radar/api/graph/basinconnectivity/BasinConnectivityGraph.java

```diff
@@ -0,0 +1,77 @@
+package cwms.radar.api.graph.basinconnectivity;
+
+import cwms.radar.api.graph.Edge;
+import cwms.radar.api.graph.Node;
+import cwms.radar.api.graph.basinconnectivity.edges.BasinConnectivityEdge;
+import cwms.radar.api.graph.basinconnectivity.nodes.BasinConnectivityNode;
+import cwms.radar.api.graph.Graph;
+import cwms.radar.data.dto.basinconnectivity.Basin;
+import cwms.radar.data.dto.basinconnectivity.Stream;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BasinConnectivityGraph implements Graph
+{
+
+    private final List<BasinConnectivityEdge> edges = new ArrayList<>();
+    private final List<BasinConnectivityNode> nodes = new ArrayList<>();
+    private final String id;
```

Comment:
i feel like we should call this "basinName" @rripken thoughts?

(but otherwise this looks good. Very simple and direct, obvious what it's suppose to be doing. I'm just starting to be really nit-picky about how things are named as it has long lasting effects.)

Reply:
I was wrong, but I thought it had to be id b/c the pg-json format wanted it to be id.  Looking at the pg-json ebnf it looks like the top level graph object is just {nodes:[...], edges:[...]} and there isn't an id field or a properties field.  Hmm,  sort of wish there was a way to name and identify and have properties for the entire graph.  
I could see "id" and "name" attributes being useful in different ways.  "basinName" is more precise about what the field holds.  Not sure how do-able it is but I'd prefer attributes that were basin-connectivity specific end up in properties[].  The structure of the connectivity format should be sort of general purpose.

Reply:
Not having an id field at the top level means that our Graph interface isn't right...

Reply:
pg-json doesn't have anything specifying what the name of the graph is really... just {nodes:[...], edges:[...]}. Seems there would need to be some sort of way of identifying what basin we are looking at though, especially if doing a getAll() ?

Reply:
I think its just a matter of naming the content-type to be clear, since including the {name:"Uppder Basin"...} isn't strictly pg-json. @MikeNeilson does application/vnd.basin-pg+json sound good?

Reply:
I like that option. The idea is we're returning a Basin, that happens to contain a graph, the graph is just part of the thing returned. But would that mean this BasinConnectivityGraph should be a member of the Basin DTO object, negating the need for an ID/name field here. This class just being "The Graph.".... I hope that made sense.

Reply:
oh and then we have a good disticntion on content

application/vnd.basin+json <- no graph, just basic basin info
application/vnd.basin-pg+json <- basin with graph

assuming we go this way doesn't have to be done now. But it seems flexible to me for the future.

Reply:
Ah. that's a discussion Ryan and I have had. _One does not need to match the database in RADAR output_. The purpose of a relation database is to store data in a standardized way to reduce duplication and increase consistency. The purpose of an API like this is to provide that data in a sensible format ready for near immediate use. Perhaps a small transform to application specific format are required; but that's why we label and maintain our output formats.

Our purpose not not to provide the normalized data set, but in fact the denormalized dataset for various use.